src/librustc_mir/borrow_check/nll/region_infer/mod.rs

```diff
@@ -444,21 +439,30 @@ impl<'tcx> RegionInferenceContext<'tcx> {
         }
     }
 
+    pub(crate) fn compute_causal_info(&self, mir: &Mir<'tcx>) -> RegionCausalInfo {
+        self.compute_causal_info2(mir, TrackCauses(true))
+    }
+
     /// Propagate the region constraints: this will grow the values
     /// for each region variable until all the constraints are
     /// satisfied. Note that some values may grow **too** large to be
     /// feasible, but we check this later.
     fn propagate_constraints(&mut self, mir: &Mir<'tcx>) {
-        debug!("propagate_constraints()");
-        debug!("propagate_constraints: constraints={:#?}", {
+        let causal_info = self.compute_causal_info2(mir, TrackCauses(false));
+        self.inferred_values = Some(causal_info.inferred_values);
+    }
+
+    fn compute_causal_info2(&self, mir: &Mir<'tcx>, track_causes: TrackCauses) -> RegionCausalInfo {
```

Comment:
I think this should return a `RegionValues`; we should build the struct in `compute_causal_info()`

Reply:
Let's rename this to `compute_region_values` or .. *anything* but something matching the regex `.*2$` =)